R/module-repository.R

```diff
@@ -42,16 +42,23 @@ remoteFileSize <- function(url) {
 #'
 #' @param name  Character string giving the module name.
 #'
-#' @param repo  GitHub repository name.
+#' @param repo  GitHub repository name, specified as \code{"username/repositoryname"}.
 #'              Default is \code{"PredictiveEcology/SpaDES-modules"}, which is
 #'              specified by the global option \code{spades.moduleRepo}.
+#'              Only \code{master} branches can be used at this point.
 #'
 #' @importFrom httr content GET stop_for_status
 #' @export
 #' @rdname getModuleVersion
 #'
+#' @details \code{getModuleVersion} extracts a module's version from the module .zip
+#'          file name. Module .zip files are searched inside in a \code{"/modules"} folder in a
+#'          GitHub public repository
```

Comment:
this isn't quite right:

1. the zip file's name is sought that matches the requested version;
2. the zipped modules are not inside the `modules/` directory -- they are inside the subdirectory for that module (`modules/myModule/myModule_x.y.z.zip`);
3. the module repo does NOT have to be a GitHub repo -- it only needs to follow the module directory structure we use.

Reply:
I'm not sure I understand point 1. In:
line 78. `zipFiles <- grep(paste0(name, "_+.+.zip"), moduleFiles, value = TRUE)`
the `grep` should match any version. Also, there doen't seem to be a "requested version" in the `getModuleVersion` function. Only module name and repo...

Also don't understand point 3.
`checkModule` need the module to be in a repo, does it not?

Reply:
Ah I see what you mean. `getModuleVersion` is simply searching for the current (most recent) version of the module, which it does by looking at the list of available zip files matching that regex. The user can request a specific version in `downloadModule`.

`checkModule` and it's friends do need the module to be in a repo, but this repo does not have to be hosted on GitHub. It can be anywhere. It just needs to be "SpaDES compatible", meaning its file and directory structure matches that of our SpaDES-modules repo on GitHub.

Reply:
Reply to your reply to point 3 (:P):
still confused about `checkModule` not needing to be hosted in GitHub. In:
line 132. `apiurl <- paste0("https://api.github.com/repos/", repo, "/git/trees/master?recursive=1")`

aren't we creating a url address that has to match a github repository?